### ISSUE TITLE
Wrong class variables tracking

### ISSUE THREAD

**Issue by socketpair** [PyCQA/pylint#4823]
### Bug description

```python
class Base:
    _class_prop: int

class Child(Base):
    _class_prop = 42

    def method(self):
        print(self._class_prop)

Child().method()
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
E1101: Instance of 'Child' has no '_class_prop' member (no-member)
```

### Expected behavior

no error

### Pylint version

```shell
pylint 2.6.0
astroid 2.4.2
Python 3.9.6 (default, Jul 16 2021, 00:00:00) 
[GCC 11.1.1 20210531 (Red Hat 11.1.1-3)]
```

**Comment by Pierre-Sassoulas** [comment]
It should be fixed in the latest pylint, you're using 2.6.0 that is kinda old now, please let me know of the problem persist, I'll reopen.

**Comment by socketpair** [comment]
@Pierre-Sassoulas did you test on latest pylint ?

**Comment by Pierre-Sassoulas** [comment]
I tested on the current main, also tested with:
```
pylint 2.9.6
astroid 2.6.6
Python 3.8.10 (default, Jun  2 2021, 10:49:15) 
[GCC 9.4.0]
```

**Comment by Pierre-Sassoulas** [comment]
Maybe you're using python 3.5 and the latest pylint you can use is 2.6.2 or your package repository do not have the latest pylint. If that's the case use pypi/pip and upgrade to python 3.6.

**Comment by socketpair** [comment]
> Maybe you're using  python 3.5

No. All of us have shown our Python versions.
